### PR TITLE
feat: add release utilities

### PR DESCRIPTION
**Pull request by bdura**
## Description

This PR ports release utilities from spaCy. These include:
- shell scripts within the `bin/` directory
- automatic git commit extraction within `setup.py`

I also added a step that extracts the library version from `setup.cfg`, to avoid repetition.

### CODE REVIEW

**Review by svlandeg — APPROVED** [review]
Nice! It wasn't easy to find a nit-pick, but I managed :trollface: 

Feel free to merge though!

File: licenses/3rd_party_licenses.txt [hunk 1]
```diff
@@ -0,0 +1,38 @@
+Third Party Licenses for Weasel
+==============================
```
[the remark this answers is not in the document]
Comment (bdura) [reply licenses/3rd_party_licenses.txt:2]: 😚 👌

File: setup.cfg [hunk 2]
```diff
@@ -1,4 +1,6 @@
 [metadata]
+name = weasel
+version = 0.1.0
```
[the remark this answers is not in the document]
Comment (bdura) [reply setup.cfg:3]: Well, it is an accepted practice: https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase

But it's inconsistent with our other projects, I'll change that.

**Comment by adrianeboyd** [comment]
The git version is related to changes in `spacy` while you're running some project, not changes in the project CLI itself, if that makes sense? (And then no third party code/licenses, which seems like it would be best to avoid for this.)

**Comment by bdura** [comment]
Oh, I guess we can use `importlib.metadata` here as well.

**Comment by adrianeboyd** [comment]
Can you explain to me why you need `__version__` for a lock file?

**Comment by bdura** [comment]
> Can you explain to me why you need `__version__` for a lock file?

The tag is a misnomer, but: https://github.com/explosion/weasel/blob/main/weasel/cli/run.py#L306

By the way, the version is _always_ checked. See [here](https://github.com/explosion/weasel/blob/main/weasel/cli/run.py#L227) and [there](https://github.com/explosion/weasel/blob/main/weasel/cli/run.py#L108): the argument is not used and always defaults to `True`.

**Comment by adrianeboyd** [comment]
Ah, then also use `importlib.metadata` in the same way to get the version for this?

But I'm not actually sure what we want to do about the spacy version in the lock files in general...

**Comment by bdura** [comment]
Yes, we should at the very least add a `weasel_version` key, and probably rename these to something else entirely for that matter, in order to avoid confusion between that and the corresponding config keys.

There are still many instances of `spacy` lurking within the code, I'm gradually making my way through.

**Comment by bdura** [comment]
I think we should change the default behaviour to _not_ re-run on a version change, by the way. Now that the project is separate from spaCy itself, I'm not sure it makes sense to re-run a computationally costly experiment should minor/patch version of the experiment manager change.

**Comment by bdura** [comment]
Converting this to draft until #15 is merged, to avoid to many conflicts.

**Comment by bdura** [comment]
Ok, that PR turned out huge - sorry about that.

I removed the git commit and version checks altogether, and added a check that the deprecated environment variables are not set. Let me know.

File: weasel/about.py [hunk 3]
```diff
@@ -1,3 +1,2 @@
-__version__ = "3.5.0"
 __projects__ = "https://github.com/explosion/projects"
 __projects_branch__ = "v3"
```
Comment (adrianeboyd) [inline weasel/about.py:2]: This will be in spacy instead, let's just remove this whole file?

File: weasel/compat.py [hunk 4]
```diff
@@ -3,3 +3,8 @@
 is_windows = sys.platform.startswith("win")
 is_linux = sys.platform.startswith("linux")
 is_osx = sys.platform == "darwin"
+
+try:
+    from importlib.metadata import version  # noqa
+except ImportError:
+    from catalogue._importlib_metadata import version  # type: ignore [no-redef]  # noqa
```
Comment (adrianeboyd) [inline weasel/compat.py:10]: If we're going to delay this to v4, then we can drop python 3.6 and 3.7 and we don't need this compat section.

File: weasel/about.py (hunk 3 above)
Comment (bdura) [reply weasel/about.py:2]: It's used for the `clone` command... We should probably remove it nonetheless, but let's do that in a separate PR?

File: weasel/compat.py (hunk 4 above)
Comment (bdura) [reply weasel/compat.py]: Done, thanks!

**Pull request merged**